crates/ruff/src/rules/flake8_copyright/settings.rs

```diff
@@ -0,0 +1,83 @@
+// //! Settings for the `flake8-copyright` plugin.
+
+use serde::{Deserialize, Serialize};
+
+use ruff_macros::{CacheKey, CombineOptions, ConfigurationOptions};
+#[derive(
+    Debug, PartialEq, Eq, Serialize, Deserialize, Default, ConfigurationOptions, CombineOptions,
+)]
+#[serde(
+    deny_unknown_fields,
+    rename_all = "kebab-case",
+    rename = "Flake8CopyrightOptions"
+)]
+#[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
+pub struct Options {
+    #[option(
+        default = "(?i)Copyright \\(C\\) \\d{4}",
+        value_type = "str",
+        example = r#"copyright-regexp = "(?i)Copyright \\(C\\) \\d{4}""#
+    )]
+    /// Regexp to check for copyright headers.
+    ///
+    /// Default is "Copyright\s+(\(C\)\s+)?\d{4}([-,]\d{4})*\s+%(author)s"
+    /// E.g # Copyright (c) 2023 ruff
+    pub copyright_regexp: Option<String>,
+    #[option(
+        default = "",
+        value_type = "str",
+        example = r#"copyright-author = "ruff""#
+    )]
+    /// Author to check for specifically within the header
+    ///
+    /// Default is ""
+    /// E.g # Copyright (c) 2023 ruff
+    pub copyright_author: Option<String>,
+    #[option(
+        default = r#"0"#,
+        value_type = "int",
+        example = r#"
+            # Ensure copyright exists within 1024 characters of header
+            copyright-min-file-size = 1024
+        "#
+    )]
+    /// Minimum number of characters in a file before requiring a copyright notice
+    pub copyright_min_file_size: Option<isize>,
+}
+
+#[derive(Debug, CacheKey)]
+pub struct Settings {
+    pub copyright_regexp: String,
+    pub copyright_author: String,
+    pub copyright_min_file_size: isize,
```

Comment:
Does the rule use `-1` to indicate no limit? 

I think I would prefer the use of an `Option` instead. 


```suggestion
    pub copyright_min_file_size: Option<u32>,
```

Reply:
converted to u32, max will be 1024 chars with default always be. rule doesn't use -1